code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -193,6 +193,12 @@
 							L[tmpname] = R
 					var/desc = input("Please select a telepad.", "RCS") in L
 					E.pad = L[desc]
+					if(!Adjacent(user))
+						to_chat(user, "<span class='notice'>Unable to teleport, too far from crate.</span>")
+						return
+					if(!(E.rcell && (E.rcell.charge >= E.chargecost)))

```

Comment:
the cost should only occur after the `do_after`, otherwise yes, you'll lose a lot of charge if you get bumped.